test/test_base.py

```diff
@@ -84,10 +84,22 @@ def start_server(self):
         # TODO: check for conflicting warnet process
         #       maybe also ensure that no conflicting docker networks exist
 
+        def write_and_print(line):
+            path = self.tmpdir / "tmp.log"
```

Comment:
no need to define this every time a log line is emitted. lets just define `self.testlog` when we instantiate the TestBase class, and open the file as a stream and write to it like a real log file. Also maybe call the file `testbase.log` instead of `tmp.log` ?

Reply:
We also have self.logfilepath which more or less is going unused.

Reply:
actually looks like that is going entirely unused, it looks like that is supposed to be the log file written by the rpc server, which I guess would have worked with docker because the server runs locally.

Reply:
I'm going to remove `self.logfilepath`.